### PR TITLE
Support IBM java

### PR DESCRIPTION
**Pull request by jamesthomp**
This change is [Reviewable](https://reviewable.io/reviews/palantir/http-remoting/430)

### CODE REVIEW

File: jaxrs-clients/src/main/java/com/palantir/remoting2/jaxrs/FeignJaxRsClientBuilder.java [hunk 1]
```diff
@@ -164,13 +164,17 @@ private static Decoder createDecoder(ObjectMapper objectMapper, ObjectMapper cbo
...
-                            : CipherSuites.fastCipherSuites())
-                    .build(),
-            ConnectionSpec.CLEARTEXT);
+        ConnectionSpec.Builder encryptedSpecs = new ConnectionSpec.Builder(ConnectionSpec.MODERN_TLS)
+                .tlsVersions(TlsVersion.TLS_1_2)
+                .cipherSuites(enableGcmCipherSuites
+                        ? CipherSuites.allCipherSuites()
+                        : CipherSuites.fastCipherSuites());
+
+        // Works around https://github.com/square/okhttp/issues/3173 until OkHttp 3.7 upgrade
```
Comment (uschi2000) [inline jaxrs-clients/src/main/java/com/palantir/remoting2/jaxrs/FeignJaxRsClientBuilder.java]: can you create an issue in http-remoting for this and link it in the comment, please?

Comment (markelliot) [reply jaxrs-clients/src/main/java/com/palantir/remoting2/jaxrs/FeignJaxRsClientBuilder.java]: what's preventing the OkHttp upgrade now?

Comment (jamesthomp) [reply jaxrs-clients/src/main/java/com/palantir/remoting2/jaxrs/FeignJaxRsClientBuilder.java]: I did try bumping to 3.8 (and 3.7), however then I ran into classpath issues (was getting NoSuchMethodException).

**Comment by jamesthomp** [comment]
closing in favor of #433

**Pull request closed without merging**